tensorboard/webapp/core/views/hash_storage_test.ts

```diff
@@ -154,13 +152,13 @@ describe('hash storage test', () => {
     expect(setPluginIdSpy).toHaveBeenCalledWith('bar', jasmine.any(Object));
   });
 
-  it('dispatches plugin changed event when hash changes', () => {
+  it('dispatches plugin changed event when popstate (hash) changes', () => {
```

Comment:
Is it possible to modify these tests so that you actually change the URL hash, including in the way that we know doesn't work for 'hashchange'?

Reply:
Considered it. Unfortunately, Karma uses actual browsers and using history API breaks the test runner but also becomes very stateful. I think this would be better tested with integrational screenshot tests.